### PR TITLE
feat(locators): add by.exactRepeater

### PR DESCRIPTION
**Pull request by sjelin**
Modified version of https://github.com/angular/protractor/pull/1662

### CODE REVIEW

File: lib/clientsidescripts.js [hunk 1]
```diff
@@ -99,7 +99,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
  * @return {Array.<Element>} The row of the repeater, or an array of elements
```
Comment (juliemr) [inline lib/clientsidescripts.js]: Update the docs to document the 'exact' parameter

File: lib/clientsidescripts.js [hunk 2]
```diff
@@ -148,7 +157,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
  *
```
Comment (juliemr) [inline lib/clientsidescripts.js]: Update the docs here as well.

File: lib/clientsidescripts.js [hunk 3]
```diff
@@ -99,7 +99,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
  * @return {Array.<Element>} The row of the repeater, or an array of elements
  *     in the first row in the case of ng-repeat-start.
  */
- functions.findRepeaterRows = function(repeater, index, using) {
+ functions.findRepeaterRows = function(repeater, exact, index, using) {
+  function repeaterMatch(ngRepeat, repeater, exact) {
+    if(exact) {
```
Comment (juliemr) [inline lib/clientsidescripts.js]: style nit: space after if

File: lib/clientsidescripts.js [hunk 4]
```diff
@@ -148,7 +157,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
  *
  * @return {Array.<Element>} All rows of the repeater.
  */
- functions.findAllRepeaterRows = function(repeater, using) {
+ functions.findAllRepeaterRows = function(repeater, exact, using) {
+  function repeaterMatch(ngRepeat, repeater, exact) {
+    if(exact) {
```
Comment (juliemr) [inline lib/clientsidescripts.js]: style nit: space after if

File: lib/clientsidescripts.js [hunk 5]
```diff
@@ -194,7 +212,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
  *
  * @return {Array.<Element>} The element in an array.
  */
-functions.findRepeaterElement = function(repeater, index, binding, using, rootSelector) {
+functions.findRepeaterElement = function(repeater, exact, index, binding, using, rootSelector) {
```
Comment (juliemr) [inline lib/clientsidescripts.js]: Ditto update docs.

File: lib/clientsidescripts.js [hunk 6]
```diff
@@ -194,7 +212,16 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
...
+  function repeaterMatch(ngRepeat, repeater, exact) {
+    if(exact) {
```
Comment (juliemr) [inline lib/clientsidescripts.js]: ditto style nit

**Comment by juliemr** [comment]
Looks good after comments.

**Comment by sjelin** [comment]
Merged in at https://github.com/angular/protractor/commit/324f69d6aa7c23ad77f1d50e26e0a56bade40132

**Pull request closed without merging**